### PR TITLE
Add listener autoconfiguration

### PR DESCRIPTION
**Pull request by jderusse**
Document the autoconfiguration for eventListener: https://github.com/symfony/symfony/pull/30760

### CODE REVIEW

File: event_dispatcher.rst [hunk 1]
```diff
@@ -129,6 +129,31 @@ listener class:
     internal Symfony listeners usually range from ``-255`` to ``255`` but your
     own listeners can use any positive or negative integer.
 
+
+.. tip::
+
+    When implementing both the ``\Symfony\Component\EventDispatcher\EventListenerInterface``
+    interface, and the ``__invoke`` method with a type-hinted arguments, Symfony
```
Comment (OskarStark) [inline event_dispatcher.rst]: s/arguments/argument/

File: event_dispatcher.rst [hunk 2]
```diff
@@ -129,6 +129,31 @@ listener class:
...
+    uses that type-hint to define the event name.
+    If :ref:`autoconfigure <services-autoconfigure>` is enabled. You neither
```
Comment (OskarStark) [inline event_dispatcher.rst]: s/. You/, you/

File: event_dispatcher.rst [hunk 3]
```diff
@@ -129,6 +129,31 @@ listener class:
...
+    don't have to manually define the ``kernel.event_listner`` tag.
+    Registering the previous listner could be simplify to:
```
Comment (OskarStark) [inline event_dispatcher.rst]: s/simplify to:/simplified to::/

And please remove the code-block:: php

File: event_dispatcher.rst [hunk 4]
```diff
@@ -129,6 +129,31 @@ listener class:
...
+
+    .. code-block:: php
+        // src/EventListener/ExceptionListener.php
+        namespace App\EventListener;
+
+        use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+
+        class ProcessException implements EventSubscriberInterface
+        {
+            public static function __invoke(GetResponseForExceptionEvent $event)
```
Comment (OskarStark) [inline event_dispatcher.rst]: If I remember correct, the GetResponseForExceptionEvent was renamed recently by @nicolas-grekas 🤔 and we should reflect that changes

**Review by OskarStark — CHANGES_REQUESTED** [review]
Thank you for your contribution 👌🏻

I left some comments

File: event_dispatcher.rst [hunk 5]
```diff
@@ -129,6 +129,31 @@ listener class:
...
+
+    When implementing both the ``\Symfony\Component\EventDispatcher\EventListenerInterface``
+    interface, and the ``__invoke`` method with a type-hinted arguments, Symfony
+    uses that type-hint to define the event name.
+    If :ref:`autoconfigure <services-autoconfigure>` is enabled. You neither
+    don't have to manually define the ``kernel.event_listner`` tag.
+    Registering the previous listner could be simplify to:
+
+    .. code-block:: php
+        // src/EventListener/ExceptionListener.php
```
Comment (OskarStark) [inline event_dispatcher.rst]: should be ProcessException.php or the class should be renamed accordingly

Comment (jderusse) [reply event_dispatcher.rst]: wrong copy/past (fixed the ClassName according the previous code block)

File: event_dispatcher.rst (hunk 4 above)
Comment (jderusse) [reply event_dispatcher.rst]: fixed. Thank you @OskarStark for that kind review

Comment (OskarStark) [reply event_dispatcher.rst]: You are very welcome, will have a look tomorrow again, I am currently on a phone 📱

File: event_dispatcher.rst [hunk 6]
```diff
@@ -129,6 +129,30 @@ listener class:
...
 
+
+.. tip::
+
+    When implementing both the ``\Symfony\Component\EventDispatcher\EventListenerInterface``
+    interface, and the ``__invoke`` method with a type-hinted argument, Symfony
+    uses that type-hint to define the event name.
+    If :ref:`autoconfigure <services-autoconfigure>` is enabled, you neither
+    don't have to manually define the ``kernel.event_listner`` tag.
+    Registering the previous listner could be simplified to:
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion
    Registering the previous listener could be simplified to::
```

Afterwards travis should be happy 💚

File: event_dispatcher.rst [hunk 7]
```diff
@@ -129,6 +129,30 @@ listener class:
     internal Symfony listeners usually range from ``-255`` to ``255`` but your
     own listeners can use any positive or negative integer.
 
+
+.. tip::
+
+    When implementing both the ``\Symfony\Component\EventDispatcher\EventListenerInterface``
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion
    When implementing both the :class:`Symfony\\Component\\EventDispatcher\\EventListenerInterface`
```

File: event_dispatcher.rst [hunk 8]
```diff
@@ -129,6 +129,30 @@ listener class:
...
+    interface, and the ``__invoke`` method with a type-hinted argument, Symfony
+    uses that type-hint to define the event name.
+    If :ref:`autoconfigure <services-autoconfigure>` is enabled, you neither
+    don't have to manually define the ``kernel.event_listner`` tag.
+    Registering the previous listner could be simplified to:
+
+        // src/EventListener/ExceptionListener.php
+        namespace App\EventListener;
+
+        use Symfony\Component\HttpKernel\Event\ExceptionEvent;
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion
        use Symfony\Component\EventDispatcher\EventListenerInterface;
        use Symfony\Component\HttpKernel\Event\ExceptionEvent;
```

File: event_dispatcher.rst [hunk 9]
```diff
@@ -129,6 +129,30 @@ listener class:
...
+
+        class ExceptionListener implements EventSubscriberInterface
+        {
+            public static function __invoke(ExceptionEvent $event)
+            {
+                // ...
+            }
+        }
+
```
Comment (OskarStark) [inline event_dispatcher.rst]:
```suggestion

```

Please remove one of the blank lines

**Review by OskarStark — APPROVED** [review]
Thank you 😊

File: event_dispatcher.rst [hunk 10]
```diff
@@ -129,6 +129,30 @@ listener class:
...
+    don't have to manually define the ``kernel.event_listner`` tag.
+    Registering the previous listener could be simplified to::
+
+        // src/EventListener/ExceptionListener.php
+        namespace App\EventListener;
+
+        use Symfony\Component\EventDispatcher\EventListenerInterface;
+        use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+
+        class ExceptionListener implements EventSubscriberInterface
```
Comment (onEXHovia) [inline event_dispatcher.rst]: `class ExceptionListener implements EventListenerInterface`

File: event_dispatcher.rst [hunk 11]
```diff
@@ -129,6 +129,30 @@ listener class:
...
+        {
+            public static function __invoke(ExceptionEvent $event)
```
Comment (onEXHovia) [inline event_dispatcher.rst]: Maybe without static?  `public function __invoke(ExceptionEvent $event)`

Comment (jderusse) [reply event_dispatcher.rst]: Thank you. I'm ashamed of having made so many mistakes :/

**Comment by javiereguiluz** [comment]
Closing because the related feature was closed without merging. Sorry!

**Pull request closed without merging**